// @azure/storage-file-share/dist/commonjs/generated/src/models/parameters.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.comp6 = exports.comp5 = exports.filePermissionFormat = exports.filePermissionKey = exports.accept2 = exports.comp4 = exports.sharePermission = exports.contentType1 = exports.comp3 = exports.breakPeriod = exports.action4 = exports.action3 = exports.action2 = exports.leaseId1 = exports.action1 = exports.requestId = exports.proposedLeaseId = exports.duration = exports.action = exports.comp2 = exports.deleteSnapshots = exports.leaseId = exports.shareSnapshot = exports.shareProvisionedBandwidthMibps = exports.shareProvisionedIops = exports.paidBurstingMaxIops = exports.paidBurstingMaxBandwidthMibps = exports.paidBurstingEnabled = exports.enableSnapshotVirtualDirectoryAccess = exports.rootSquash = exports.enabledProtocols = exports.accessTier = exports.quota = exports.metadata = exports.restype1 = exports.include = exports.maxResults = exports.marker = exports.prefix = exports.comp1 = exports.accept1 = exports.fileRequestIntent = exports.version = exports.timeoutInSeconds = exports.comp = exports.restype = exports.url = exports.accept = exports.properties = exports.contentType = void 0;
exports.range1 = exports.comp12 = exports.accept3 = exports.body = exports.contentType2 = exports.fileContentLength1 = exports.rangeGetContentMD5 = exports.range = exports.nfsFileType = exports.fileContentDisposition = exports.fileContentMD5 = exports.fileCacheControl = exports.fileContentLanguage = exports.fileContentEncoding = exports.fileContentType = exports.fileTypeConstant = exports.fileContentLength = exports.allowSourceTrailingDot = exports.fileChangeTime = exports.fileLastWriteTime = exports.fileCreationTime = exports.fileAttributes1 = exports.destinationLeaseId = exports.sourceLeaseId = exports.ignoreReadOnly = exports.replaceIfExists = exports.renameSource = exports.comp11 = exports.handleId = exports.comp10 = exports.recursive = exports.comp9 = exports.includeExtendedInfo = exports.include1 = exports.fileMode = exports.group = exports.owner = exports.fileChangeOn = exports.fileLastWriteOn = exports.fileCreatedOn = exports.fileAttributes = exports.filePermissionKey1 = exports.filePermission = exports.allowTrailingDot = exports.restype2 = exports.deletedShareVersion = exports.deletedShareName = exports.comp8 = exports.comp7 = exports.shareAcl = void 0;
exports.targetFile = exports.restype4 = exports.linkText = exports.restype3 = exports.copyActionAbortConstant = exports.copyId = exports.comp14 = exports.fileOwnerCopyMode = exports.fileModeCopyMode = exports.setArchiveAttribute = exports.ignoreReadOnly1 = exports.filePermissionCopyMode = exports.supportRename = exports.prevsharesnapshot = exports.comp13 = exports.copySourceAuthorization = exports.sourceIfNoneMatchCrc64 = exports.sourceIfMatchCrc64 = exports.sourceContentCrc64 = exports.fileRangeWriteFromUrl = exports.sourceRange = exports.copySource = exports.fileLastWrittenMode = exports.contentMD5 = exports.contentLength = exports.fileRangeWrite = void 0;
const mappers_js_1 = require("../models/mappers.js");
exports.contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/xml",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
exports.properties = {
    parameterPath: "properties",
    mapper: mappers_js_1.FileServiceProperties,
};
exports.accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/xml",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.url = {
    parameterPath: "url",
    mapper: {
        serializedName: "url",
        required: true,
        xmlName: "url",
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
exports.restype = {
    parameterPath: "restype",
    mapper: {
        defaultValue: "service",
        isConstant: true,
        serializedName: "restype",
        type: {
            name: "String",
        },
    },
};
exports.comp = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "properties",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.timeoutInSeconds = {
    parameterPath: ["options", "timeoutInSeconds"],
    mapper: {
        constraints: {
            InclusiveMinimum: 0,
        },
        serializedName: "timeout",
        xmlName: "timeout",
        type: {
            name: "Number",
        },
    },
};
exports.version = {
    parameterPath: "version",
    mapper: {
        defaultValue: "2025-11-05",
        isConstant: true,
        serializedName: "x-ms-version",
        type: {
            name: "String",
        },
    },
};
exports.fileRequestIntent = {
    parameterPath: ["options", "fileRequestIntent"],
    mapper: {
        serializedName: "x-ms-file-request-intent",
        xmlName: "x-ms-file-request-intent",
        type: {
            name: "String",
        },
    },
};
exports.accept1 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/xml",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.comp1 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "list",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.prefix = {
    parameterPath: ["options", "prefix"],
    mapper: {
        serializedName: "prefix",
        xmlName: "prefix",
        type: {
            name: "String",
        },
    },
};
exports.marker = {
    parameterPath: ["options", "marker"],
    mapper: {
        serializedName: "marker",
        xmlName: "marker",
        type: {
            name: "String",
        },
    },
};
exports.maxResults = {
    parameterPath: ["options", "maxResults"],
    mapper: {
        constraints: {
            InclusiveMinimum: 1,
        },
        serializedName: "maxresults",
        xmlName: "maxresults",
        type: {
            name: "Number",
        },
    },
};
exports.include = {
    parameterPath: ["options", "include"],
    mapper: {
        serializedName: "include",
        xmlName: "include",
        xmlElementName: "ListSharesIncludeType",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Enum",
                    allowedValues: ["snapshots", "metadata", "deleted"],
                },
            },
        },
    },
    collectionFormat: "CSV",
};
exports.restype1 = {
    parameterPath: "restype",
    mapper: {
        defaultValue: "share",
        isConstant: true,
        serializedName: "restype",
        type: {
            name: "String",
        },
    },
};
exports.metadata = {
    parameterPath: ["options", "metadata"],
    mapper: {
        serializedName: "x-ms-meta",
        xmlName: "x-ms-meta",
        headerCollectionPrefix: "x-ms-meta-",
        type: {
            name: "Dictionary",
            value: { type: { name: "String" } },
        },
    },
};
exports.quota = {
    parameterPath: ["options", "quota"],
    mapper: {
        constraints: {
            InclusiveMinimum: 1,
        },
        serializedName: "x-ms-share-quota",
        xmlName: "x-ms-share-quota",
        type: {
            name: "Number",
        },
    },
};
exports.accessTier = {
    parameterPath: ["options", "accessTier"],
    mapper: {
        serializedName: "x-ms-access-tier",
        xmlName: "x-ms-access-tier",
        type: {
            name: "Enum",
            allowedValues: ["TransactionOptimized", "Hot", "Cool", "Premium"],
        },
    },
};
exports.enabledProtocols = {
    parameterPath: ["options", "enabledProtocols"],
    mapper: {
        serializedName: "x-ms-enabled-protocols",
        xmlName: "x-ms-enabled-protocols",
        type: {
            name: "String",
        },
    },
};
exports.rootSquash = {
    parameterPath: ["options", "rootSquash"],
    mapper: {
        serializedName: "x-ms-root-squash",
        xmlName: "x-ms-root-squash",
        type: {
            name: "Enum",
            allowedValues: ["NoRootSquash", "RootSquash", "AllSquash"],
        },
    },
};
exports.enableSnapshotVirtualDirectoryAccess = {
    parameterPath: ["options", "enableSnapshotVirtualDirectoryAccess"],
    mapper: {
        serializedName: "x-ms-enable-snapshot-virtual-directory-access",
        xmlName: "x-ms-enable-snapshot-virtual-directory-access",
        type: {
            name: "Boolean",
        },
    },
};
exports.paidBurstingEnabled = {
    parameterPath: ["options", "paidBurstingEnabled"],
    mapper: {
        serializedName: "x-ms-share-paid-bursting-enabled",
        xmlName: "x-ms-share-paid-bursting-enabled",
        type: {
            name: "Boolean",
        },
    },
};
exports.paidBurstingMaxBandwidthMibps = {
    parameterPath: ["options", "paidBurstingMaxBandwidthMibps"],
    mapper: {
        serializedName: "x-ms-share-paid-bursting-max-bandwidth-mibps",
        xmlName: "x-ms-share-paid-bursting-max-bandwidth-mibps",
        type: {
            name: "Number",
        },
    },
};
exports.paidBurstingMaxIops = {
    parameterPath: ["options", "paidBurstingMaxIops"],
    mapper: {
        serializedName: "x-ms-share-paid-bursting-max-iops",
        xmlName: "x-ms-share-paid-bursting-max-iops",
        type: {
            name: "Number",
        },
    },
};
exports.shareProvisionedIops = {
    parameterPath: ["options", "shareProvisionedIops"],
    mapper: {
        serializedName: "x-ms-share-provisioned-iops",
        xmlName: "x-ms-share-provisioned-iops",
        type: {
            name: "Number",
        },
    },
};
exports.shareProvisionedBandwidthMibps = {
    parameterPath: ["options", "shareProvisionedBandwidthMibps"],
    mapper: {
        serializedName: "x-ms-share-provisioned-bandwidth-mibps",
        xmlName: "x-ms-share-provisioned-bandwidth-mibps",
        type: {
            name: "Number",
        },
    },
};
exports.shareSnapshot = {
    parameterPath: ["options", "shareSnapshot"],
    mapper: {
        serializedName: "sharesnapshot",
        xmlName: "sharesnapshot",
        type: {
            name: "String",
        },
    },
};
exports.leaseId = {
    parameterPath: ["options", "leaseAccessConditions", "leaseId"],
    mapper: {
        serializedName: "x-ms-lease-id",
        xmlName: "x-ms-lease-id",
        type: {
            name: "String",
        },
    },
};
exports.deleteSnapshots = {
    parameterPath: ["options", "deleteSnapshots"],
    mapper: {
        serializedName: "x-ms-delete-snapshots",
        xmlName: "x-ms-delete-snapshots",
        type: {
            name: "Enum",
            allowedValues: ["include", "include-leased"],
        },
    },
};
exports.comp2 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "lease",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.action = {
    parameterPath: "action",
    mapper: {
        defaultValue: "acquire",
        isConstant: true,
        serializedName: "x-ms-lease-action",
        type: {
            name: "String",
        },
    },
};
exports.duration = {
    parameterPath: ["options", "duration"],
    mapper: {
        serializedName: "x-ms-lease-duration",
        xmlName: "x-ms-lease-duration",
        type: {
            name: "Number",
        },
    },
};
exports.proposedLeaseId = {
    parameterPath: ["options", "proposedLeaseId"],
    mapper: {
        serializedName: "x-ms-proposed-lease-id",
        xmlName: "x-ms-proposed-lease-id",
        type: {
            name: "String",
        },
    },
};
exports.requestId = {
    parameterPath: ["options", "requestId"],
    mapper: {
        serializedName: "x-ms-client-request-id",
        xmlName: "x-ms-client-request-id",
        type: {
            name: "String",
        },
    },
};
exports.action1 = {
    parameterPath: "action",
    mapper: {
        defaultValue: "release",
        isConstant: true,
        serializedName: "x-ms-lease-action",
        type: {
            name: "String",
        },
    },
};
exports.leaseId1 = {
    parameterPath: "leaseId",
    mapper: {
        serializedName: "x-ms-lease-id",
        required: true,
        xmlName: "x-ms-lease-id",
        type: {
            name: "String",
        },
    },
};
exports.action2 = {
    parameterPath: "action",
    mapper: {
        defaultValue: "change",
        isConstant: true,
        serializedName: "x-ms-lease-action",
        type: {
            name: "String",
        },
    },
};
exports.action3 = {
    parameterPath: "action",
    mapper: {
        defaultValue: "renew",
        isConstant: true,
        serializedName: "x-ms-lease-action",
        type: {
            name: "String",
        },
    },
};
exports.action4 = {
    parameterPath: "action",
    mapper: {
        defaultValue: "break",
        isConstant: true,
        serializedName: "x-ms-lease-action",
        type: {
            name: "String",
        },
    },
};
exports.breakPeriod = {
    parameterPath: ["options", "breakPeriod"],
    mapper: {
        serializedName: "x-ms-lease-break-period",
        xmlName: "x-ms-lease-break-period",
        type: {
            name: "Number",
        },
    },
};
exports.comp3 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "snapshot",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.contentType1 = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
exports.sharePermission = {
    parameterPath: "sharePermission",
    mapper: mappers_js_1.SharePermission,
};
exports.comp4 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "filepermission",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.accept2 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.filePermissionKey = {
    parameterPath: "filePermissionKey",
    mapper: {
        serializedName: "x-ms-file-permission-key",
        required: true,
        xmlName: "x-ms-file-permission-key",
        type: {
            name: "String",
        },
    },
};
exports.filePermissionFormat = {
    parameterPath: ["options", "filePermissionFormat"],
    mapper: {
        serializedName: "x-ms-file-permission-format",
        xmlName: "x-ms-file-permission-format",
        type: {
            name: "Enum",
            allowedValues: ["Sddl", "Binary"],
        },
    },
};
exports.comp5 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "metadata",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.comp6 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "acl",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.shareAcl = {
    parameterPath: ["options", "shareAcl"],
    mapper: {
        serializedName: "shareAcl",
        xmlName: "SignedIdentifiers",
        xmlIsWrapped: true,
        xmlElementName: "SignedIdentifier",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Composite",
                    className: "SignedIdentifier",
                },
            },
        },
    },
};
exports.comp7 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "stats",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.comp8 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "undelete",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.deletedShareName = {
    parameterPath: ["options", "deletedShareName"],
    mapper: {
        serializedName: "x-ms-deleted-share-name",
        xmlName: "x-ms-deleted-share-name",
        type: {
            name: "String",
        },
    },
};
exports.deletedShareVersion = {
    parameterPath: ["options", "deletedShareVersion"],
    mapper: {
        serializedName: "x-ms-deleted-share-version",
        xmlName: "x-ms-deleted-share-version",
        type: {
            name: "String",
        },
    },
};
exports.restype2 = {
    parameterPath: "restype",
    mapper: {
        defaultValue: "directory",
        isConstant: true,
        serializedName: "restype",
        type: {
            name: "String",
        },
    },
};
exports.allowTrailingDot = {
    parameterPath: ["options", "allowTrailingDot"],
    mapper: {
        serializedName: "x-ms-allow-trailing-dot",
        xmlName: "x-ms-allow-trailing-dot",
        type: {
            name: "Boolean",
        },
    },
};
exports.filePermission = {
    parameterPath: ["options", "filePermission"],
    mapper: {
        serializedName: "x-ms-file-permission",
        xmlName: "x-ms-file-permission",
        type: {
            name: "String",
        },
    },
};
exports.filePermissionKey1 = {
    parameterPath: ["options", "filePermissionKey"],
    mapper: {
        serializedName: "x-ms-file-permission-key",
        xmlName: "x-ms-file-permission-key",
        type: {
            name: "String",
        },
    },
};
exports.fileAttributes = {
    parameterPath: ["options", "fileAttributes"],
    mapper: {
        serializedName: "x-ms-file-attributes",
        xmlName: "x-ms-file-attributes",
        type: {
            name: "String",
        },
    },
};
exports.fileCreatedOn = {
    parameterPath: ["options", "fileCreatedOn"],
    mapper: {
        serializedName: "x-ms-file-creation-time",
        xmlName: "x-ms-file-creation-time",
        type: {
            name: "String",
        },
    },
};
exports.fileLastWriteOn = {
    parameterPath: ["options", "fileLastWriteOn"],
    mapper: {
        serializedName: "x-ms-file-last-write-time",
        xmlName: "x-ms-file-last-write-time",
        type: {
            name: "String",
        },
    },
};
exports.fileChangeOn = {
    parameterPath: ["options", "fileChangeOn"],
    mapper: {
        serializedName: "x-ms-file-change-time",
        xmlName: "x-ms-file-change-time",
        type: {
            name: "String",
        },
    },
};
exports.owner = {
    parameterPath: ["options", "owner"],
    mapper: {
        serializedName: "x-ms-owner",
        xmlName: "x-ms-owner",
        type: {
            name: "String",
        },
    },
};
exports.group = {
    parameterPath: ["options", "group"],
    mapper: {
        serializedName: "x-ms-group",
        xmlName: "x-ms-group",
        type: {
            name: "String",
        },
    },
};
exports.fileMode = {
    parameterPath: ["options", "fileMode"],
    mapper: {
        serializedName: "x-ms-mode",
        xmlName: "x-ms-mode",
        type: {
            name: "String",
        },
    },
};
exports.include1 = {
    parameterPath: ["options", "include"],
    mapper: {
        serializedName: "include",
        xmlName: "include",
        xmlElementName: "ListFilesIncludeType",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Enum",
                    allowedValues: ["Timestamps", "Etag", "Attributes", "PermissionKey"],
                },
            },
        },
    },
    collectionFormat: "CSV",
};
exports.includeExtendedInfo = {
    parameterPath: ["options", "includeExtendedInfo"],
    mapper: {
        serializedName: "x-ms-file-extended-info",
        xmlName: "x-ms-file-extended-info",
        type: {
            name: "Boolean",
        },
    },
};
exports.comp9 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "listhandles",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.recursive = {
    parameterPath: ["options", "recursive"],
    mapper: {
        serializedName: "x-ms-recursive",
        xmlName: "x-ms-recursive",
        type: {
            name: "Boolean",
        },
    },
};
exports.comp10 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "forceclosehandles",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.handleId = {
    parameterPath: "handleId",
    mapper: {
        serializedName: "x-ms-handle-id",
        required: true,
        xmlName: "x-ms-handle-id",
        type: {
            name: "String",
        },
    },
};
exports.comp11 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "rename",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.renameSource = {
    parameterPath: "renameSource",
    mapper: {
        serializedName: "x-ms-file-rename-source",
        required: true,
        xmlName: "x-ms-file-rename-source",
        type: {
            name: "String",
        },
    },
};
exports.replaceIfExists = {
    parameterPath: ["options", "replaceIfExists"],
    mapper: {
        serializedName: "x-ms-file-rename-replace-if-exists",
        xmlName: "x-ms-file-rename-replace-if-exists",
        type: {
            name: "Boolean",
        },
    },
};
exports.ignoreReadOnly = {
    parameterPath: ["options", "ignoreReadOnly"],
    mapper: {
        serializedName: "x-ms-file-rename-ignore-readonly",
        xmlName: "x-ms-file-rename-ignore-readonly",
        type: {
            name: "Boolean",
        },
    },
};
exports.sourceLeaseId = {
    parameterPath: ["options", "sourceLeaseAccessConditions", "sourceLeaseId"],
    mapper: {
        serializedName: "x-ms-source-lease-id",
        xmlName: "x-ms-source-lease-id",
        type: {
            name: "String",
        },
    },
};
exports.destinationLeaseId = {
    parameterPath: [
        "options",
        "destinationLeaseAccessConditions",
        "destinationLeaseId",
    ],
    mapper: {
        serializedName: "x-ms-destination-lease-id",
        xmlName: "x-ms-destination-lease-id",
        type: {
            name: "String",
        },
    },
};
exports.fileAttributes1 = {
    parameterPath: ["options", "copyFileSmbInfo", "fileAttributes"],
    mapper: {
        serializedName: "x-ms-file-attributes",
        xmlName: "x-ms-file-attributes",
        type: {
            name: "String",
        },
    },
};
exports.fileCreationTime = {
    parameterPath: ["options", "copyFileSmbInfo", "fileCreationTime"],
    mapper: {
        serializedName: "x-ms-file-creation-time",
        xmlName: "x-ms-file-creation-time",
        type: {
            name: "String",
        },
    },
};
exports.fileLastWriteTime = {
    parameterPath: ["options", "copyFileSmbInfo", "fileLastWriteTime"],
    mapper: {
        serializedName: "x-ms-file-last-write-time",
        xmlName: "x-ms-file-last-write-time",
        type: {
            name: "String",
        },
    },
};
exports.fileChangeTime = {
    parameterPath: ["options", "copyFileSmbInfo", "fileChangeTime"],
    mapper: {
        serializedName: "x-ms-file-change-time",
        xmlName: "x-ms-file-change-time",
        type: {
            name: "String",
        },
    },
};
exports.allowSourceTrailingDot = {
    parameterPath: ["options", "allowSourceTrailingDot"],
    mapper: {
        serializedName: "x-ms-source-allow-trailing-dot",
        xmlName: "x-ms-source-allow-trailing-dot",
        type: {
            name: "Boolean",
        },
    },
};
exports.fileContentLength = {
    parameterPath: "fileContentLength",
    mapper: {
        serializedName: "x-ms-content-length",
        required: true,
        xmlName: "x-ms-content-length",
        type: {
            name: "Number",
        },
    },
};
exports.fileTypeConstant = {
    parameterPath: "fileTypeConstant",
    mapper: {
        defaultValue: "file",
        isConstant: true,
        serializedName: "x-ms-type",
        type: {
            name: "String",
        },
    },
};
exports.fileContentType = {
    parameterPath: ["options", "fileHttpHeaders", "fileContentType"],
    mapper: {
        serializedName: "x-ms-content-type",
        xmlName: "x-ms-content-type",
        type: {
            name: "String",
        },
    },
};
exports.fileContentEncoding = {
    parameterPath: ["options", "fileHttpHeaders", "fileContentEncoding"],
    mapper: {
        serializedName: "x-ms-content-encoding",
        xmlName: "x-ms-content-encoding",
        type: {
            name: "String",
        },
    },
};
exports.fileContentLanguage = {
    parameterPath: ["options", "fileHttpHeaders", "fileContentLanguage"],
    mapper: {
        serializedName: "x-ms-content-language",
        xmlName: "x-ms-content-language",
        type: {
            name: "String",
        },
    },
};
exports.fileCacheControl = {
    parameterPath: ["options", "fileHttpHeaders", "fileCacheControl"],
    mapper: {
        serializedName: "x-ms-cache-control",
        xmlName: "x-ms-cache-control",
        type: {
            name: "String",
        },
    },
};
exports.fileContentMD5 = {
    parameterPath: ["options", "fileHttpHeaders", "fileContentMD5"],
    mapper: {
        serializedName: "x-ms-content-md5",
        xmlName: "x-ms-content-md5",
        type: {
            name: "ByteArray",
        },
    },
};
exports.fileContentDisposition = {
    parameterPath: ["options", "fileHttpHeaders", "fileContentDisposition"],
    mapper: {
        serializedName: "x-ms-content-disposition",
        xmlName: "x-ms-content-disposition",
        type: {
            name: "String",
        },
    },
};
exports.nfsFileType = {
    parameterPath: ["options", "nfsFileType"],
    mapper: {
        serializedName: "x-ms-file-file-type",
        xmlName: "x-ms-file-file-type",
        type: {
            name: "String",
        },
    },
};
exports.range = {
    parameterPath: ["options", "range"],
    mapper: {
        serializedName: "x-ms-range",
        xmlName: "x-ms-range",
        type: {
            name: "String",
        },
    },
};
exports.rangeGetContentMD5 = {
    parameterPath: ["options", "rangeGetContentMD5"],
    mapper: {
        serializedName: "x-ms-range-get-content-md5",
        xmlName: "x-ms-range-get-content-md5",
        type: {
            name: "Boolean",
        },
    },
};
exports.fileContentLength1 = {
    parameterPath: ["options", "fileContentLength"],
    mapper: {
        serializedName: "x-ms-content-length",
        xmlName: "x-ms-content-length",
        type: {
            name: "Number",
        },
    },
};
exports.contentType2 = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/octet-stream",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
exports.body = {
    parameterPath: ["options", "body"],
    mapper: {
        serializedName: "body",
        xmlName: "body",
        type: {
            name: "Stream",
        },
    },
};
exports.accept3 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/xml",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.comp12 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "range",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.range1 = {
    parameterPath: "range",
    mapper: {
        serializedName: "x-ms-range",
        required: true,
        xmlName: "x-ms-range",
        type: {
            name: "String",
        },
    },
};
exports.fileRangeWrite = {
    parameterPath: "fileRangeWrite",
    mapper: {
        defaultValue: "update",
        serializedName: "x-ms-write",
        required: true,
        xmlName: "x-ms-write",
        type: {
            name: "Enum",
            allowedValues: ["update", "clear"],
        },
    },
};
exports.contentLength = {
    parameterPath: "contentLength",
    mapper: {
        serializedName: "Content-Length",
        required: true,
        xmlName: "Content-Length",
        type: {
            name: "Number",
        },
    },
};
exports.contentMD5 = {
    parameterPath: ["options", "contentMD5"],
    mapper: {
        serializedName: "Content-MD5",
        xmlName: "Content-MD5",
        type: {
            name: "ByteArray",
        },
    },
};
exports.fileLastWrittenMode = {
    parameterPath: ["options", "fileLastWrittenMode"],
    mapper: {
        serializedName: "x-ms-file-last-write-time",
        xmlName: "x-ms-file-last-write-time",
        type: {
            name: "Enum",
            allowedValues: ["Now", "Preserve"],
        },
    },
};
exports.copySource = {
    parameterPath: "copySource",
    mapper: {
        serializedName: "x-ms-copy-source",
        required: true,
        xmlName: "x-ms-copy-source",
        type: {
            name: "String",
        },
    },
};
exports.sourceRange = {
    parameterPath: ["options", "sourceRange"],
    mapper: {
        serializedName: "x-ms-source-range",
        xmlName: "x-ms-source-range",
        type: {
            name: "String",
        },
    },
};
exports.fileRangeWriteFromUrl = {
    parameterPath: "fileRangeWriteFromUrl",
    mapper: {
        defaultValue: "update",
        isConstant: true,
        serializedName: "x-ms-write",
        type: {
            name: "String",
        },
    },
};
exports.sourceContentCrc64 = {
    parameterPath: ["options", "sourceContentCrc64"],
    mapper: {
        serializedName: "x-ms-source-content-crc64",
        xmlName: "x-ms-source-content-crc64",
        type: {
            name: "ByteArray",
        },
    },
};
exports.sourceIfMatchCrc64 = {
    parameterPath: [
        "options",
        "sourceModifiedAccessConditions",
        "sourceIfMatchCrc64",
    ],
    mapper: {
        serializedName: "x-ms-source-if-match-crc64",
        xmlName: "x-ms-source-if-match-crc64",
        type: {
            name: "ByteArray",
        },
    },
};
exports.sourceIfNoneMatchCrc64 = {
    parameterPath: [
        "options",
        "sourceModifiedAccessConditions",
        "sourceIfNoneMatchCrc64",
    ],
    mapper: {
        serializedName: "x-ms-source-if-none-match-crc64",
        xmlName: "x-ms-source-if-none-match-crc64",
        type: {
            name: "ByteArray",
        },
    },
};
exports.copySourceAuthorization = {
    parameterPath: ["options", "copySourceAuthorization"],
    mapper: {
        serializedName: "x-ms-copy-source-authorization",
        xmlName: "x-ms-copy-source-authorization",
        type: {
            name: "String",
        },
    },
};
exports.comp13 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "rangelist",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.prevsharesnapshot = {
    parameterPath: ["options", "prevsharesnapshot"],
    mapper: {
        serializedName: "prevsharesnapshot",
        xmlName: "prevsharesnapshot",
        type: {
            name: "String",
        },
    },
};
exports.supportRename = {
    parameterPath: ["options", "supportRename"],
    mapper: {
        serializedName: "x-ms-file-support-rename",
        xmlName: "x-ms-file-support-rename",
        type: {
            name: "Boolean",
        },
    },
};
exports.filePermissionCopyMode = {
    parameterPath: ["options", "copyFileSmbInfo", "filePermissionCopyMode"],
    mapper: {
        serializedName: "x-ms-file-permission-copy-mode",
        xmlName: "x-ms-file-permission-copy-mode",
        type: {
            name: "Enum",
            allowedValues: ["source", "override"],
        },
    },
};
exports.ignoreReadOnly1 = {
    parameterPath: ["options", "copyFileSmbInfo", "ignoreReadOnly"],
    mapper: {
        serializedName: "x-ms-file-copy-ignore-readonly",
        xmlName: "x-ms-file-copy-ignore-readonly",
        type: {
            name: "Boolean",
        },
    },
};
exports.setArchiveAttribute = {
    parameterPath: ["options", "copyFileSmbInfo", "setArchiveAttribute"],
    mapper: {
        serializedName: "x-ms-file-copy-set-archive",
        xmlName: "x-ms-file-copy-set-archive",
        type: {
            name: "Boolean",
        },
    },
};
exports.fileModeCopyMode = {
    parameterPath: ["options", "fileModeCopyMode"],
    mapper: {
        serializedName: "x-ms-file-mode-copy-mode",
        xmlName: "x-ms-file-mode-copy-mode",
        type: {
            name: "Enum",
            allowedValues: ["source", "override"],
        },
    },
};
exports.fileOwnerCopyMode = {
    parameterPath: ["options", "fileOwnerCopyMode"],
    mapper: {
        serializedName: "x-ms-file-owner-copy-mode",
        xmlName: "x-ms-file-owner-copy-mode",
        type: {
            name: "Enum",
            allowedValues: ["source", "override"],
        },
    },
};
exports.comp14 = {
    parameterPath: "comp",
    mapper: {
        defaultValue: "copy",
        isConstant: true,
        serializedName: "comp",
        type: {
            name: "String",
        },
    },
};
exports.copyId = {
    parameterPath: "copyId",
    mapper: {
        serializedName: "copyid",
        required: true,
        xmlName: "copyid",
        type: {
            name: "String",
        },
    },
};
exports.copyActionAbortConstant = {
    parameterPath: "copyActionAbortConstant",
    mapper: {
        defaultValue: "abort",
        isConstant: true,
        serializedName: "x-ms-copy-action",
        type: {
            name: "String",
        },
    },
};
exports.restype3 = {
    parameterPath: "restype",
    mapper: {
        defaultValue: "symboliclink",
        isConstant: true,
        serializedName: "restype",
        type: {
            name: "String",
        },
    },
};
exports.linkText = {
    parameterPath: "linkText",
    mapper: {
        serializedName: "x-ms-link-text",
        required: true,
        xmlName: "x-ms-link-text",
        type: {
            name: "String",
        },
    },
};
exports.restype4 = {
    parameterPath: "restype",
    mapper: {
        defaultValue: "hardlink",
        isConstant: true,
        serializedName: "restype",
        type: {
            name: "String",
        },
    },
};
exports.targetFile = {
    parameterPath: "targetFile",
    mapper: {
        serializedName: "x-ms-file-target-file",
        required: true,
        xmlName: "x-ms-file-target-file",
        type: {
            name: "String",
        },
    },
};
