// @azure/storage-file-share/dist/react-native/generated/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ShareTokenIntent} that the service accepts. */
export var KnownShareTokenIntent;
(function (KnownShareTokenIntent) {
    /** Backup */
    KnownShareTokenIntent["Backup"] = "backup";
})(KnownShareTokenIntent || (KnownShareTokenIntent = {}));
/** Known values of {@link NfsFileType} that the service accepts. */
export var KnownNfsFileType;
(function (KnownNfsFileType) {
    /** Regular */
    KnownNfsFileType["Regular"] = "Regular";
    /** Directory */
    KnownNfsFileType["Directory"] = "Directory";
    /** SymLink */
    KnownNfsFileType["SymLink"] = "SymLink";
})(KnownNfsFileType || (KnownNfsFileType = {}));
/** Known values of {@link StorageErrorCode} that the service accepts. */
export var KnownStorageErrorCode;
(function (KnownStorageErrorCode) {
    /** AccountAlreadyExists */
    KnownStorageErrorCode["AccountAlreadyExists"] = "AccountAlreadyExists";
    /** AccountBeingCreated */
    KnownStorageErrorCode["AccountBeingCreated"] = "AccountBeingCreated";
    /** AccountIsDisabled */
    KnownStorageErrorCode["AccountIsDisabled"] = "AccountIsDisabled";
    /** AuthenticationFailed */
    KnownStorageErrorCode["AuthenticationFailed"] = "AuthenticationFailed";
    /** AuthorizationFailure */
    KnownStorageErrorCode["AuthorizationFailure"] = "AuthorizationFailure";
    /** ConditionHeadersNotSupported */
    KnownStorageErrorCode["ConditionHeadersNotSupported"] = "ConditionHeadersNotSupported";
    /** ConditionNotMet */
    KnownStorageErrorCode["ConditionNotMet"] = "ConditionNotMet";
    /** EmptyMetadataKey */
    KnownStorageErrorCode["EmptyMetadataKey"] = "EmptyMetadataKey";
    /** FileShareProvisionedBandwidthDowngradeNotAllowed */
    KnownStorageErrorCode["FileShareProvisionedBandwidthDowngradeNotAllowed"] = "FileShareProvisionedBandwidthDowngradeNotAllowed";
    /** FileShareProvisionedIopsDowngradeNotAllowed */
    KnownStorageErrorCode["FileShareProvisionedIopsDowngradeNotAllowed"] = "FileShareProvisionedIopsDowngradeNotAllowed";
    /** InsufficientAccountPermissions */
    KnownStorageErrorCode["InsufficientAccountPermissions"] = "InsufficientAccountPermissions";
    /** InternalError */
    KnownStorageErrorCode["InternalError"] = "InternalError";
    /** InvalidAuthenticationInfo */
    KnownStorageErrorCode["InvalidAuthenticationInfo"] = "InvalidAuthenticationInfo";
    /** InvalidHeaderValue */
    KnownStorageErrorCode["InvalidHeaderValue"] = "InvalidHeaderValue";
    /** InvalidHttpVerb */
    KnownStorageErrorCode["InvalidHttpVerb"] = "InvalidHttpVerb";
    /** InvalidInput */
    KnownStorageErrorCode["InvalidInput"] = "InvalidInput";
    /** InvalidMd5 */
    KnownStorageErrorCode["InvalidMd5"] = "InvalidMd5";
    /** InvalidMetadata */
    KnownStorageErrorCode["InvalidMetadata"] = "InvalidMetadata";
    /** InvalidQueryParameterValue */
    KnownStorageErrorCode["InvalidQueryParameterValue"] = "InvalidQueryParameterValue";
    /** InvalidRange */
    KnownStorageErrorCode["InvalidRange"] = "InvalidRange";
    /** InvalidResourceName */
    KnownStorageErrorCode["InvalidResourceName"] = "InvalidResourceName";
    /** InvalidUri */
    KnownStorageErrorCode["InvalidUri"] = "InvalidUri";
    /** InvalidXmlDocument */
    KnownStorageErrorCode["InvalidXmlDocument"] = "InvalidXmlDocument";
    /** InvalidXmlNodeValue */
    KnownStorageErrorCode["InvalidXmlNodeValue"] = "InvalidXmlNodeValue";
    /** Md5Mismatch */
    KnownStorageErrorCode["Md5Mismatch"] = "Md5Mismatch";
    /** MetadataTooLarge */
    KnownStorageErrorCode["MetadataTooLarge"] = "MetadataTooLarge";
    /** MissingContentLengthHeader */
    KnownStorageErrorCode["MissingContentLengthHeader"] = "MissingContentLengthHeader";
    /** MissingRequiredQueryParameter */
    KnownStorageErrorCode["MissingRequiredQueryParameter"] = "MissingRequiredQueryParameter";
    /** MissingRequiredHeader */
    KnownStorageErrorCode["MissingRequiredHeader"] = "MissingRequiredHeader";
    /** MissingRequiredXmlNode */
    KnownStorageErrorCode["MissingRequiredXmlNode"] = "MissingRequiredXmlNode";
    /** MultipleConditionHeadersNotSupported */
    KnownStorageErrorCode["MultipleConditionHeadersNotSupported"] = "MultipleConditionHeadersNotSupported";
    /** OperationTimedOut */
    KnownStorageErrorCode["OperationTimedOut"] = "OperationTimedOut";
    /** OutOfRangeInput */
    KnownStorageErrorCode["OutOfRangeInput"] = "OutOfRangeInput";
    /** OutOfRangeQueryParameterValue */
    KnownStorageErrorCode["OutOfRangeQueryParameterValue"] = "OutOfRangeQueryParameterValue";
    /** RequestBodyTooLarge */
    KnownStorageErrorCode["RequestBodyTooLarge"] = "RequestBodyTooLarge";
    /** ResourceTypeMismatch */
    KnownStorageErrorCode["ResourceTypeMismatch"] = "ResourceTypeMismatch";
    /** RequestUrlFailedToParse */
    KnownStorageErrorCode["RequestUrlFailedToParse"] = "RequestUrlFailedToParse";
    /** ResourceAlreadyExists */
    KnownStorageErrorCode["ResourceAlreadyExists"] = "ResourceAlreadyExists";
    /** ResourceNotFound */
    KnownStorageErrorCode["ResourceNotFound"] = "ResourceNotFound";
    /** ServerBusy */
    KnownStorageErrorCode["ServerBusy"] = "ServerBusy";
    /** UnsupportedHeader */
    KnownStorageErrorCode["UnsupportedHeader"] = "UnsupportedHeader";
    /** UnsupportedXmlNode */
    KnownStorageErrorCode["UnsupportedXmlNode"] = "UnsupportedXmlNode";
    /** UnsupportedQueryParameter */
    KnownStorageErrorCode["UnsupportedQueryParameter"] = "UnsupportedQueryParameter";
    /** UnsupportedHttpVerb */
    KnownStorageErrorCode["UnsupportedHttpVerb"] = "UnsupportedHttpVerb";
    /** CannotDeleteFileOrDirectory */
    KnownStorageErrorCode["CannotDeleteFileOrDirectory"] = "CannotDeleteFileOrDirectory";
    /** ClientCacheFlushDelay */
    KnownStorageErrorCode["ClientCacheFlushDelay"] = "ClientCacheFlushDelay";
    /** DeletePending */
    KnownStorageErrorCode["DeletePending"] = "DeletePending";
    /** DirectoryNotEmpty */
    KnownStorageErrorCode["DirectoryNotEmpty"] = "DirectoryNotEmpty";
    /** FileLockConflict */
    KnownStorageErrorCode["FileLockConflict"] = "FileLockConflict";
    /** InvalidFileOrDirectoryPathName */
    KnownStorageErrorCode["InvalidFileOrDirectoryPathName"] = "InvalidFileOrDirectoryPathName";
    /** ParentNotFound */
    KnownStorageErrorCode["ParentNotFound"] = "ParentNotFound";
    /** ReadOnlyAttribute */
    KnownStorageErrorCode["ReadOnlyAttribute"] = "ReadOnlyAttribute";
    /** ShareAlreadyExists */
    KnownStorageErrorCode["ShareAlreadyExists"] = "ShareAlreadyExists";
    /** ShareBeingDeleted */
    KnownStorageErrorCode["ShareBeingDeleted"] = "ShareBeingDeleted";
    /** ShareDisabled */
    KnownStorageErrorCode["ShareDisabled"] = "ShareDisabled";
    /** ShareNotFound */
    KnownStorageErrorCode["ShareNotFound"] = "ShareNotFound";
    /** SharingViolation */
    KnownStorageErrorCode["SharingViolation"] = "SharingViolation";
    /** ShareSnapshotInProgress */
    KnownStorageErrorCode["ShareSnapshotInProgress"] = "ShareSnapshotInProgress";
    /** ShareSnapshotCountExceeded */
    KnownStorageErrorCode["ShareSnapshotCountExceeded"] = "ShareSnapshotCountExceeded";
    /** ShareSnapshotOperationNotSupported */
    KnownStorageErrorCode["ShareSnapshotOperationNotSupported"] = "ShareSnapshotOperationNotSupported";
    /** ShareHasSnapshots */
    KnownStorageErrorCode["ShareHasSnapshots"] = "ShareHasSnapshots";
    /** PreviousSnapshotNotFound */
    KnownStorageErrorCode["PreviousSnapshotNotFound"] = "PreviousSnapshotNotFound";
    /** ContainerQuotaDowngradeNotAllowed */
    KnownStorageErrorCode["ContainerQuotaDowngradeNotAllowed"] = "ContainerQuotaDowngradeNotAllowed";
    /** AuthorizationSourceIPMismatch */
    KnownStorageErrorCode["AuthorizationSourceIPMismatch"] = "AuthorizationSourceIPMismatch";
    /** AuthorizationProtocolMismatch */
    KnownStorageErrorCode["AuthorizationProtocolMismatch"] = "AuthorizationProtocolMismatch";
    /** AuthorizationPermissionMismatch */
    KnownStorageErrorCode["AuthorizationPermissionMismatch"] = "AuthorizationPermissionMismatch";
    /** AuthorizationServiceMismatch */
    KnownStorageErrorCode["AuthorizationServiceMismatch"] = "AuthorizationServiceMismatch";
    /** AuthorizationResourceTypeMismatch */
    KnownStorageErrorCode["AuthorizationResourceTypeMismatch"] = "AuthorizationResourceTypeMismatch";
    /** FeatureVersionMismatch */
    KnownStorageErrorCode["FeatureVersionMismatch"] = "FeatureVersionMismatch";
    /** ShareSnapshotNotFound */
    KnownStorageErrorCode["ShareSnapshotNotFound"] = "ShareSnapshotNotFound";
})(KnownStorageErrorCode || (KnownStorageErrorCode = {}));
